index.js

```diff
@@ -171,6 +171,10 @@ var compile = function(schema, cache, root, reporter, opts) {
       }
     }
 
+    if (node === false) {
```

Comment:
In addition to failing early with `false` schemas, we could potentially save a significant amount of work by passing early for `true` schemas. I'm not familiar enough with the codebase to know whether that could be significant or not, but keep it in mind.

Reply:
It's a great point, I'm just scared of falling into the performance rabbit hole. I already started slipping, and wrote https://github.com/deBhal/is-my-json-valid/pull/2 by way of backstop :) 

Let's continue this discussion there.